### PR TITLE
Fix failure when controller methods are private

### PR DESCRIPTION
**Pull request by RobinDaugherty**
The `webhook_secret` method should be private, but the calls to `respond_to?` did not include the second parameter which would look for a private method. This fixes it and updates the readme to give a private method in the example.

### CODE REVIEW

File: lib/github_webhook/processor.rb [hunk 1]
```diff
@@ -64,7 +64,7 @@ class UnsupportedContentTypeError < StandardError; end
   )
 
   def create
-    if self.respond_to? event_method
+    if self.respond_to?(event_method, true)
```
Comment (ssaunier) [inline lib/github_webhook/processor.rb]: So basically you're saying that'd you rather have all the `github_` methods private, do I understand that correctly? Because this line is not related to `:webhook_secret`.

File: spec/github_webhook/processor_spec.rb [hunk 2]
```diff
@@ -28,25 +28,49 @@ def github_push(payload)
       end
     end
 
+    class ControllerWithPrivateSecret < ControllerWithoutSecret
```
Comment (ssaunier) [inline spec/github_webhook/processor_spec.rb]: I like the fact that you are not breaking existing behavior! 👏  👍

File: lib/github_webhook/processor.rb (hunk 1 above)
Comment (RobinDaugherty) [reply lib/github_webhook/processor.rb]: You're right, this affects the event method instead, but since it would to have it be private as well, I figured it would make sense to change this to work with private methods as well.
